File: app/src/main/java/ru/magflayer/spectrum/presentation/common/mvp/view/PageView.java
package ru.magflayer.spectrum.presentation.common.mvp.view;


import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import ru.magflayer.spectrum.presentation.common.mvp.strategy.AddToEndSingleByTagStateStrategy;

public interface PageView extends BaseView {

    String PROGRESS_TAG = "progress_tag";

    @StateStrategyType(value = AddToEndSingleByTagStateStrategy.class, tag = PROGRESS_TAG)
    void showProgressBar();

    @StateStrategyType(value = AddToEndSingleByTagStateStrategy.class, tag = PROGRESS_TAG)
    void hideProgressBar();

}

File: app/src/main/java/ru/magflayer/spectrum/presentation/common/mvp/view/PageView.kt
package ru.magflayer.spectrum.presentation.common.mvp.view

import moxy.viewstate.strategy.AddToEndSingleTagStrategy
import moxy.viewstate.strategy.StateStrategyType

interface PageView : BaseView {

    companion object {
        const val PROGRESS_TAG = "progress_tag"
    }

    @StateStrategyType(value = AddToEndSingleTagStrategy::class, tag = PROGRESS_TAG)
    fun showProgressBar()

    @StateStrategyType(value = AddToEndSingleTagStrategy::class, tag = PROGRESS_TAG)
    fun hideProgressBar()
}
